lib/rom/repository/changeset/stateful.rb

```diff
@@ -18,7 +18,7 @@ class Stateful < Changeset
       # @!attribute [r] pipe
       #   @return [Changeset::Pipe] data transformation pipe
       #   @api private
-      option :pipe, accept: [Proc, Pipe], default: -> { self.class.default_pipe(self) }
```

Comment:
At this point, a `option :dep` from a subclass isn't set, so `self`, which is becoming the execution context of the `map` block, doesn't have the dependency available. So I think you have to initialize `@pipe` manually after `self` is fully set up.